core/controllers/base.py

```diff
@@ -184,6 +184,8 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
         self.user = current_user_services.get_current_user()
         self.user_id = current_user_services.get_user_id(
             self.user) if self.user else None
+        self.role = user_services.get_user_role_from_id(
```

Comment:
You should be able to trace the import chain. current_user_services is imported at the top of this file.

In this case, it is imported from the platform module which delegates to gae_current_user_services.py, which is a file in our codebase. I recommend reading it to see what it does, and referring to the GAE documentation if necessary.